WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java

```diff
@@ -207,6 +208,10 @@ public static ReaderPost fromJson(JSONObject json) {
             post.setRailcarJson(jsonRailcar.toString());
         }
 
+        // set the card type last since it depends on information contained in the post - note
+        // that this is stored in the post table rather than calculated on-the-fly
```

Comment:
Thanks a lot for this comment. Quick question, is the `fromJson` handled in a background thread? If not, running this multiple times might be a costly operation especially due to stripping of text for the calculation.